extensions/mssql/src/objectExplorerNodeProvider/fileSources.ts

```diff
@@ -329,29 +238,32 @@ export class HdfsFileSource implements IFileSource {
 			let fileName = fspath.basename(localFile.path);
 			let remotePath = joinHdfsPath(remoteDirPath, fileName);
 
-			let writeStream = this.client.createWriteStream(remotePath);
-
-			let readStream = fs.createReadStream(localFile.path);
-			readStream.pipe(writeStream);
-
-			let error: string | Error = undefined;
```

Comment:
It is not hard-casting. `<HdfsError>` does not do anything, but preventing syntax error.
Like my previous comment, the err object is `HdfsError` type originally because I attached error handler to fs.WriteStream in the `createWriteStream()` implementation as well, so it emits `HdfsError` type error (the return value of `this.parseError()`):
Following code shows error handler that emits `HdfsError` is attached to the `writeStream`.
```
...
let req = request(params, (error, response, body) => {
    // Handle redirect only if there was not an error (e.g. res is defined)
    if (response && this.isRedirect(response)) {
        let upload = request(
            Object.assign(params, { url: response.headers.location }),
            (err, res, bo) => {
                if (err || this.isError(res)) {
                    emitError(req, this.parseError(res, bo, err));
                    req.end();
                } else if (res.headers.hasOwnProperty('location')) {
                    req.emit('finish', res.headers.location);
                } else {
                    req.emit('finish');
                }
            }
        );
        canResume = true; // Enable resume
        stream.pipe(upload);
        stream.resume();
    }

    if (error || this.isError(response)) {
        emitError(req, this.parseError(response, body, error));
    }
});
...
```
I already tested and confirmed the `err` is in `HdfsError` type for this as well.

Again, if we use `Error`, `string`, or `any` type, then we potentially lose the benefit of using `HdfsError` which provide this additional information:
`statusCode`, `statusMessage`, `remoteExceptionMessage`, original `error` object.
This information would be lost and not useable in the higher stream.